sql/migrations/20210810191400_world.sql

```diff
@@ -617,6 +617,27 @@ UPDATE `creature_template` SET `gossip_menu_id` = 5081, `npc_flags` = 3 WHERE `e
 INSERT INTO `gossip_menu` (`entry`, `text_id`) VALUES
 (5081, 6288);
 
+-- Seersa Copperpinch, Guchie Jinglepocket and Jaycrue Copperpinch
+UPDATE `creature_template` SET `gossip_menu_id` = 5181 WHERE `entry` IN (13432, 13436, 13430);
+UPDATE `creature_template` SET `npc_flags` = 7 WHERE `entry` = 13432;
+UPDATE `creature_template` SET `npc_flags` = 5 WHERE `entry` = 13436;
+UPDATE `creature_template` SET `npc_flags` = 5 WHERE `entry` = 13430;
+
+-- Scarshield Infiltrator
+UPDATE `creature_template` SET `gossip_menu_id` = 2630, `npc_flags` = 1 WHERE `entry`=10299;

```

Comment:
This npc changes entry before you can talk to him.

Reply:
And npc_flags are 0 for it in sniff, so why are you changing them to 1?

Reply:
@ratkosrb doesnt the npc have to have that flag set if it has a gossip menu?

Reply:
This npc does NOT have gossip. If the reason you are adding gossip to it is because you see gossip menus for its entry inside the `creature_gossip` table in the sniff db, you are missing out on a key fact about how that works. The entry in this table, is the entry that is part of the creature's guid, for which a gossip menu was sent. The entry from the guid is the original entry with which it spawns. This value does not change. This npc however can never be interacted with in its normal form. As you can see in the sniffs it has npc_flags = 0 which means you can't talk to it. When a player goes near the npc and has the Unadorned Seal of Ascension item in his inventory, the npc change's its entry to Vaelan (10296), which you can see in the EventAI script for the npc. And actually the menu already exists and is fully implemented, but its using a placeholder id (above 10k). All you need to do is update the existing placeholder gossip menu ids for Vaelan to the ones in the sniff. You can see that they start from 12039. Look in the gossip menu table in vmangos and match them with the real ids from sniffs.

Example of what you have to do:
```
UPDATE `creature_template` SET `gossip_menu_id`=2630 WHERE `gossip_menu_id`=12039;
UPDATE `gossip_menu` SET `entry`=2630 WHERE `entry`=12039;
UPDATE `gossip_menu_option` SET `menu_id`=2630 WHERE `menu_id`=12039;
UPDATE `gossip_menu_option` SET `action_menu_id`=2630 WHERE `action_menu_id`=12039;
```